LobbyServer2/LobbyServer/Stats/StatsApi.cs

```diff
@@ -0,0 +1,276 @@
+                }
+            }
+
+            string teamWin = gameSummary.GameResult == GameResult.TeamAWon ? "TeamA" : "TeamB";
+            Guid guid = Guid.NewGuid();
+            string guidString = guid.ToString("N");
+            string numericString = string.Concat(guidString.Select(c => ((int)c).ToString("D3")));
+            // Take the first 19 numbers
+            string first19Numbers = new(numericString.Take(19).ToArray());
+            string gameId = first19Numbers; // Bot will change this later to messageId from discord so linking in launcher works but for now we need an unique id we have GameServerProcessCode to change all this
+            string score = $"{gameSummary.TeamAPoints}-{gameSummary.TeamBPoints}";
+            string turns = gameSummary.NumOfTurns.ToString();
+            string GameServerProcessCode = gameInfo.GameServerProcessCode;
+
+            var gameData = new
+            {
+                data = new
+                {
+                    date = DateTime.UtcNow.ToString("o"), // ISO 8601 format
+                    gameid = gameId,
+                    teamwin = teamWin,
+                    turns,
+                    score,
+                    map,
+                    gametype = gameType,
+                    server = serverName,
+                    version = serverVersion,
+                    GameServerProcessCode,
+                    channelid = "" // bot will set this we dont have it,
+                }
+            };
+
+            string json = JsonConvert.SerializeObject(gameData);
+            StringContent content = new(json, Encoding.UTF8, "application/json");
+
+            HttpResponseMessage response = await client.PostAsync($"{Get().conf.ApiUrl}/games", content);
+
+            if (response.IsSuccessStatusCode)
+            {
+                string responseBody = await response.Content.ReadAsStringAsync();
+                var responseJson = JsonConvert.DeserializeObject<JObject>(responseBody);
+                string id = responseJson["data"]["id"].ToString();
+
+                // Post player stats
+                foreach (PlayerGameSummary player in gameSummary.PlayerGameSummaryList)
+                {
+                    string handle;
+                    if (player.AccountId != 0)
+                    {
+                        PersistedAccountData account = DB.Get().AccountDao.GetAccount(player.AccountId);
+                        handle = account?.Handle ?? "UNKNOWN";
+                    }
+                    else
+                    {
+                        handle = player.CharacterName;
+                    }
+
+                    bool Deadliest = false;
+                    bool Supportiest = false;
+                    bool Tankiest = false;
+                    bool MostDecorated = false;
+
+                    BadgeAndParticipantInfo badges = gameSummary.BadgeAndParticipantsInfo
+                        .FirstOrDefault(p => p.PlayerId == player.PlayerId);
```

Comment:
You probably want to match by ActorIndex. If one player controls multiple characters, they are going to have the same PlayerId.